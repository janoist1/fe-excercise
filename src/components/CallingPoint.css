.CallingPoint {
  display: flex;
  height: 50px;
}

.CallingPoint.-departed .CallingPoint__scheduled-at,
.CallingPoint.-departed .CallingPoint__station {
  color: gray;
}

.CallingPoint__scheduled-at {
  font-size: 18px;
}

.CallingPoint__indicator {
  position: relative;
  margin: 0 15px 0 22px;
  border-left: 1px solid gray;
}

.CallingPoint__indicator::before {
  content: '';
  margin-left: -7px;
  display: block;
  width: 11px;
  height: 11px;
  border-radius: 100%;
  border: 1px solid gray;
  background-color: white;
}

.CallingPoint__indicator.-origin::before,
.CallingPoint__indicator.-destination::before {
  margin-left: -9px;
  width: 15px;
  height: 15px;
  border-radius: 100%;
  background-color: black;
}

.CallingPoint__indicator.-destination {
  border: 0;
}

.CallingPoint__train {
  position: absolute;
  top: 0;
  left: -12px;
  width: 21px;
  height: 21px;
  border-radius: 100%;
  border: 1px solid gray;
  background-color: cyan;
}

.CallingPoint__train.-departed {
  top: 20px;
}

.CallingPoint__train::before {
  content: 'ðŸš†';
  margin-left: 2.5px;
}

.CallingPoint__status {
  margin: 0;
}

.CallingPoint__station {
  font-weight: 700;
}

.CallingPoint__estimated-at {
  font-size: 14px;
  color: gray;
}
