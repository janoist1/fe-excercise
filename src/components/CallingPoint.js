import React from 'react';
import './CallingPoint.css';

const getIndicatorClasses = ({ hasDeparted, isTrainHere, isOrigin, isDestination }) => {
  const classes = ['CallingPoint__indicator'];

  hasDeparted && classes.push('-departed');
  isTrainHere && classes.push('-train-here');
  isOrigin && classes.push('-origin');
  isDestination && classes.push('-destination');

  return classes.join(' ');
};

const CallingPoint = ({ hasDeparted, isTrainHere, station, scheduledAt, estimatedAt, isOrigin, isDestination }) => (
  <div className={`CallingPoint${hasDeparted ? ' -departed' : ''}`}>
    <time className="CallingPoint__scheduled-at">{ scheduledAt }</time>
    <div className={getIndicatorClasses({ hasDeparted, isTrainHere, isOrigin, isDestination })}>
      {isTrainHere && <div className={`CallingPoint__train${hasDeparted ? ' -departed' : ''}`} />}
    </div>
    <dl className="CallingPoint__status">
      <dt className="CallingPoint__station">{ station }</dt>
      <dd className="CallingPoint__estimated-at">{ `${hasDeparted ? 'Dept.' : 'Exp.'} ${estimatedAt}` }</dd>
    </dl>
  </div>
);

export default CallingPoint;
