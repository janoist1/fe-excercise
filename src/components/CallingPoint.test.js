import { shallow } from 'enzyme';
import React from 'react';
import CallingPoint from './CallingPoint';

describe('<CallingPoint />', () => {
  it('renders all the variations correctly', () => {
    const renderComponent = props => shallow(<CallingPoint {...props} />);

    expect(
      renderComponent({
        hasDeparted: false,
        isTrainHere: false,
        station: 'Sevenoaks',
        scheduledAt: '10:00',
        estimatedAt: 'On time',
        isOrigin: false,
        isDestination: false
      }).html()
    ).toMatchSnapshot();

    expect(
      renderComponent({
        hasDeparted: true,
        isTrainHere: false,
        station: 'Sevenoaks',
        scheduledAt: '10:00',
        estimatedAt: 'On time',
        isOrigin: false,
        isDestination: false
      }).html()
    ).toMatchSnapshot();

    expect(
      renderComponent({
        hasDeparted: false,
        isTrainHere: true,
        station: 'Sevenoaks',
        scheduledAt: '10:00',
        estimatedAt: 'On time',
        isOrigin: false,
        isDestination: false
      }).html()
    ).toMatchSnapshot();

    expect(
      renderComponent({
        hasDeparted: true,
        isTrainHere: true,
        station: 'Sevenoaks',
        scheduledAt: '10:00',
        estimatedAt: 'On time',
        isOrigin: false,
        isDestination: false
      }).html()
    ).toMatchSnapshot();

    expect(
      renderComponent({
        hasDeparted: false,
        isTrainHere: false,
        station: 'Sevenoaks',
        scheduledAt: '10:00',
        estimatedAt: 'On time',
        isOrigin: true,
        isDestination: false
      }).html()
    ).toMatchSnapshot();

    expect(
      renderComponent({
        hasDeparted: false,
        isTrainHere: false,
        station: 'Sevenoaks',
        scheduledAt: '10:00',
        estimatedAt: 'On time',
        isOrigin: false,
        isDestination: true
      }).html()
    ).toMatchSnapshot();
  });
});
