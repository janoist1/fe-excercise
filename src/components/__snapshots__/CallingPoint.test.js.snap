exports[`<CallingPoint /> renders all the variations correctly 1`] = `"<div class=\"CallingPoint\"><time class=\"CallingPoint__scheduled-at\">10:00</time><div class=\"CallingPoint__indicator\"></div><dl class=\"CallingPoint__status\"><dt class=\"CallingPoint__station\">Sevenoaks</dt><dd class=\"CallingPoint__estimated-at\">Exp. On time</dd></dl></div>"`;

exports[`<CallingPoint /> renders all the variations correctly 2`] = `"<div class=\"CallingPoint -departed\"><time class=\"CallingPoint__scheduled-at\">10:00</time><div class=\"CallingPoint__indicator -departed\"></div><dl class=\"CallingPoint__status\"><dt class=\"CallingPoint__station\">Sevenoaks</dt><dd class=\"CallingPoint__estimated-at\">Dept. On time</dd></dl></div>"`;

exports[`<CallingPoint /> renders all the variations correctly 3`] = `"<div class=\"CallingPoint\"><time class=\"CallingPoint__scheduled-at\">10:00</time><div class=\"CallingPoint__indicator -train-here\"><div class=\"CallingPoint__train\"></div></div><dl class=\"CallingPoint__status\"><dt class=\"CallingPoint__station\">Sevenoaks</dt><dd class=\"CallingPoint__estimated-at\">Exp. On time</dd></dl></div>"`;

exports[`<CallingPoint /> renders all the variations correctly 4`] = `"<div class=\"CallingPoint -departed\"><time class=\"CallingPoint__scheduled-at\">10:00</time><div class=\"CallingPoint__indicator -departed -train-here\"><div class=\"CallingPoint__train -departed\"></div></div><dl class=\"CallingPoint__status\"><dt class=\"CallingPoint__station\">Sevenoaks</dt><dd class=\"CallingPoint__estimated-at\">Dept. On time</dd></dl></div>"`;

exports[`<CallingPoint /> renders all the variations correctly 5`] = `"<div class=\"CallingPoint\"><time class=\"CallingPoint__scheduled-at\">10:00</time><div class=\"CallingPoint__indicator -origin\"></div><dl class=\"CallingPoint__status\"><dt class=\"CallingPoint__station\">Sevenoaks</dt><dd class=\"CallingPoint__estimated-at\">Exp. On time</dd></dl></div>"`;

exports[`<CallingPoint /> renders all the variations correctly 6`] = `"<div class=\"CallingPoint\"><time class=\"CallingPoint__scheduled-at\">10:00</time><div class=\"CallingPoint__indicator -destination\"></div><dl class=\"CallingPoint__status\"><dt class=\"CallingPoint__station\">Sevenoaks</dt><dd class=\"CallingPoint__estimated-at\">Exp. On time</dd></dl></div>"`;
